cg/meta/demultiplex/utils.py

```diff
@@ -35,27 +36,32 @@ def get_lane_from_sample_fastq(sample_fastq_path: Path) -> int:
     raise ValueError(f"Could not extract lane number from fastq file name {sample_fastq_path.name}")
 
 
-def get_sample_id_from_sample_fastq(sample_fastq: Path) -> str:
-    """
-    Extract sample id from fastq file path.
+def get_sample_fastqs_from_flow_cell(
+    flow_cell_directory: Path, sample_internal_id: str
+) -> Optional[List[Path]]:
+    """Retrieve sample FastQs from a flow cell directory."""
+    root_pattern = f"{sample_internal_id}_S*_L*_R*_*{FileExtensions.FASTQ}{FileExtensions.GZIP}"
+    unaligned_pattern = f"Unaligned*/Project_*/Sample_{sample_internal_id}/*{FileExtensions.FASTQ}{FileExtensions.GZIP}"
 
-    Pre-condition:
-        - The parent directory name contains the sample id formatted as Sample_<sample_id>
-    """
+    for pattern in [root_pattern, unaligned_pattern]:
+        sample_fastqs: List[Path] = get_files_matching_pattern(
+            directory=flow_cell_directory, pattern=pattern
+        )
+        valid_sample_fastqs: List[Path] = get_valid_sample_fastqs(
+            fastq_paths=sample_fastqs, sample_internal_id=sample_internal_id
+        )
 
-    sample_parent_match = re.search(r"Sample_(\w+)", sample_fastq.parent.name)
+        if valid_sample_fastqs:
+            return valid_sample_fastqs
```

Comment:
Second this.

Reply:
Good point! Finally a comment not related to naming 😅

Reply:
Since this function is in a utils module, I prefer letting the caller handle the case where no files were found. It currently logs a warning message that no files could be found for the specific sample: 

```python
sample_fastq_paths: Optional[List[Path]] = get_sample_fastqs_from_flow_cell(
    flow_cell_directory=flow_cell.path, sample_internal_id=sample_internal_id
)

if not sample_fastq_paths:
    LOG.warning(
        f"Cannot find fastq files for sample {sample_internal_id} in {flow_cell.path}. Skipping."
    )
    continue
```

Reply:
I prefer this over raising an exception since that increases the verbosity of the code. Not sure what the best practice is